### PR TITLE
[ERM UI] Update release content page to match mock ups

### PR DESCRIPTION
**Pull request by gherceg**
## Summary

Straggler PR I forgot about. My local env is totally borked right now, otherwise I would attach a screenshot. This is mainly wording changes, as well as adding text to the checkbox on the Push Content page. I know I ran this locally when my local env worked fine. I'll add a screenshot once I fix my issue, but wanted to create the PR in the meantime.

Screen Shot 2021-06-18 at 9 37 03 AM

## Feature Flag

`LINKED_DOMAINS`
## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### QA Plan

Will run through QA on parent branch.
### Safety story

Locally tested at some point. Cannot locally test right now.
### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/linked_domain/templates/linked_domain/tabs/push_release_content_tab.html [hunk 1]
```diff
@@ -2,6 +2,14 @@
 {% load i18n %}
 
 <div id="ko-tabs-push-content" class="ko-template">
+  <h3>{% trans "Push Content" %}</h3>
+  <p>
+    {% blocktrans %}
+      Prepare a new release by selecting which data models you would like to push to the selected downstream project spaces.<br/>
+      <a href="#" target="_blank">Learn more</a> about pushing release content with linked projects.
```
Comment (orangejenny) [inline corehq/apps/linked_domain/templates/linked_domain/tabs/push_release_content_tab.html:9]: confirming that these docs aren't ready yet?

File: corehq/apps/linked_domain/templates/linked_domain/tabs/push_release_content_tab.html [hunk 2]
```diff
@@ -43,14 +51,13 @@
     </div>
     <div class="form-group">
       <label class="col-sm-3 col-md-2 control-label" for="build-apps">
-        {% trans "Create and release new build for apps" %}
+        {% trans "Re-Build Applications" %}
```
Comment (orangejenny) [inline corehq/apps/linked_domain/templates/linked_domain/tabs/push_release_content_tab.html:54]: `Re-Build` => `Rebuild`?

File: corehq/apps/linked_domain/templates/linked_domain/tabs/pull_release_content_tab.html [hunk 3]
```diff
@@ -5,7 +5,7 @@
   <div data-bind="if: domainLink">
     <p>
       {% blocktrans %}
-        This project is linked to the upstream project <a data-bind="attr: {'href': domainLink.upstreamUrl}, text: domainLink.master_domain"></a>.<br/>
+        This project is linked to the upstream project <a data-bind="attr: {'href': domainLink.upstreamUrl}, text: domainLink.upstreamDomain"></a>.<br/>
```
Comment (orangejenny) [inline corehq/apps/linked_domain/templates/linked_domain/tabs/pull_release_content_tab.html:8]: Is this change supposed to be part of this PR? I looked at the version of `domain_links.js` that's on the `gh/improve-erm-ui` which is [this](https://github.com/dimagi/commcare-hq/blob/e83c1984be750d0499a7dd4ddd8be9ea94a3013b/corehq/apps/linked_domain/static/linked_domain/js/domain_links.js), and it still has `master_domain`.

**Review by orangejenny — APPROVED** [review]
Fine to merge if the `upstreamDomain` change is intended.

File: corehq/apps/linked_domain/templates/linked_domain/tabs/pull_release_content_tab.html (hunk 3 above)
Comment (gherceg) [reply corehq/apps/linked_domain/templates/linked_domain/tabs/pull_release_content_tab.html:8]: Good catch. I messed up a conflict resolution and thought I had taken care of it, but obviously not.

File: corehq/apps/linked_domain/templates/linked_domain/tabs/push_release_content_tab.html (hunk 1 above)
Comment (gherceg) [reply corehq/apps/linked_domain/templates/linked_domain/tabs/push_release_content_tab.html:9]: Yeah pretty sure public facing docs are not ready. I'll make a ticket to make sure I address these.

**Comment by gherceg** [comment]
Updated with screenshot and added another commit to replace "release" with "push". If we stick with this wording I will rename the underlying variables to match the updated lingo.

**Pull request merged**